extensions/service-catalog/model/src/main/resources/schema/servicecatalog-schema.json

```diff
@@ -158,15 +158,15 @@
           "required": true
         },
         "metadata": {
-          "javaType": "io.fabric8.kubernetes.api.model.ObjectMeta"
+          "existingJavaType": "io.fabric8.kubernetes.api.model.ObjectMeta"
         },
         "spec": {
           "$ref": "#/definitions/github_com_kubernetes-sigs_service-catalog_pkg_apis_servicecatalog_v1beta1_ClusterServiceBrokerSpec",
-          "javaType": "io.fabric8.servicecatalog.api.model.ClusterServiceBrokerSpec"
+          "existingJavaType": "io.fabric8.servicecatalog.api.model.ClusterServiceBrokerSpec"
```

Comment:
Isn't this class (and a lot others) supposed to be generated?

Reply:
I checked and it looks like it's still generated:
```
kubernetes-client : $ find . -iname ClusterServiceBrokerSpec.java
./extensions/service-catalog/model/target/generated-sources/io/fabric8/servicecatalog/api/model/ClusterServiceBrokerSpec.java
```
Looks like `javaType` is split into `existingJavaType` and `javaType` in this version of JsonSchema2Pojo[0]

[0] https://github.com/joelittlejohn/jsonschema2pojo/issues/828

Reply:
Yes, the class is meant to be generated.

In order to generate **only** the necessary classes, some are set as `javaType` and others as `existingJavaType`.

So for modules that export classes, you will find in the "definitions" section those schemas.
For this particular class, this would be here:
https://github.com/fabric8io/kubernetes-client/blob/4eee56eb02cc3880435bcf54aacd86c735205013/extensions/service-catalog/model/src/main/resources/schema/servicecatalog-schema.json#L229
https://github.com/fabric8io/kubernetes-client/blob/4eee56eb02cc3880435bcf54aacd86c735205013/extensions/service-catalog/model/src/main/resources/schema/servicecatalog-schema.json#L265

However, that class might be used as a field/property in a different type (the line we're currently commenting on).

This is where things get complicated, especially for those modules that reference classes from other modules.

So the hard part and core implementation of this PR is figuring out which definitions must be declared as  `javaType` or `existingJavaType`

For example, in the OpenShift model, there are many top-level definitions that belong to the core module:
https://github.com/fabric8io/kubernetes-client/blob/4eee56eb02cc3880435bcf54aacd86c735205013/kubernetes-model-generator/openshift-model/src/main/resources/schema/kube-schema.json#L6

thus, must be declared as existing despite being top-level definitions:
https://github.com/fabric8io/kubernetes-client/blob/4eee56eb02cc3880435bcf54aacd86c735205013/kubernetes-model-generator/openshift-model/src/main/resources/schema/kube-schema.json#L49